pkg/deploy/lattice/listener_synthesizer.go

```diff
@@ -80,6 +89,45 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
 	return nil
 }
 
+func (l *listenerSynthesizer) getLatticeListenerDefaultAction(ctx context.Context, stackListener *model.Listener) (
+	*vpclattice.RuleAction, error,
+) {
+	if stackListener.Spec.DefaultAction == nil ||
+		(stackListener.Spec.DefaultAction.FixedResponseStatusCode == nil && stackListener.Spec.DefaultAction.Forward == nil) {
+		return nil, fmt.Errorf("invalid listener default action, must be either fixed response or forward")
+	}
+
+	if stackListener.Spec.DefaultAction.FixedResponseStatusCode != nil {
+		return &vpclattice.RuleAction{
+			FixedResponse: &vpclattice.FixedResponseAction{
+				StatusCode: stackListener.Spec.DefaultAction.FixedResponseStatusCode,
+			},
+		}, nil
+	}
+
+	// If the listener DefaultAction is not fixed response, for example for TLS_PASSTHROUGH listener, it must be a forward action, fill the forward action target group ids for it
+	if err := l.tgManager.ResolveRuleTgIds(ctx, stackListener.Spec.DefaultAction.Forward, l.stack); err != nil {
+		return nil, fmt.Errorf("failed to resolve rule tg ids, err = %v", err)
+	}
```

Comment:
Bring this block into the `Synthesize()` method.

Reply:
Will change in the new revision